install-validator/internal/dir/compare.go

```diff
@@ -0,0 +1,70 @@
+package dir
+
+import (
+	"bytes"
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/vmware-tanzu/graph-framework-for-microservices/install-validator/internal/kubernetes"
+
+	"sigs.k8s.io/yaml"
+
+	nexuscompare "github.com/vmware-tanzu/graph-framework-for-microservices/common-library/pkg/nexus-compare"
+)
+
+type compareFunc func([]byte, []byte) (bool, *bytes.Buffer, error)
+
+func CheckDir(dir string, c kubernetes.ClientInt, cFunc compareFunc) ([]string, []string, *bytes.Buffer, error) {
+	var changes []*bytes.Buffer
+	var incNames []string
```

Comment:
Instead of storing the incompatible CRD name in one list and the error message in other list, what I suspect you are looking for is a map[string]*bytes.Buffer
where key is the CRD name and value is the text.

Keeping them in two different list essentially makes them unrelated.